src/charm.py

```diff
@@ -481,159 +44,46 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
-        )
-        dedup_ingress_relations = []
-        for relation in ingress_relations:
-            if self._gen_relation_dedup_key(relation) in nginx_route_relation_keys:
-                LOGGER.warning(
-                    "legacy ingress relation from app %s is shadowed by nginx-route relation",
-                    relation.app,
-                )
-                continue
-            dedup_ingress_relations.append(relation)
-        return tuple(nginx_route_relations + dedup_ingress_relations)
-
-    @property
-    def _all_config_or_relations(self) -> Any:
-        """Get all configuration and relation data.
-
-        Returns:
-            All configuration and relation data.
-        """
-        all_relations = self._deduped_relations()
-        if not all_relations:
-            all_relations = (None,)  # type: ignore[assignment]
-        multiple_rels = self._multiple_relations
-        return [
-            _ConfigOrRelation(self.model, self.config, relation, multiple_rels)
-            for relation in all_relations
-        ]
-
-    @property
-    def _multiple_relations(self) -> bool:
-        """Check if we're related to multiple applications.
-
-        Returns:
-            if we're related to multiple applications or not.
-        """
-        return len(self._deduped_relations()) > 1
-
-    @property
-    def _namespace(self) -> Any:
-        """Namespace for this ingress.
-
-        Returns:
-            The namespace for this Ingress.
-        """
-        # We're querying the first one here because this will always be the same
-        # for all instances. It would be very unusual for a relation to specify
-        # this (arguably we should remove this as a relation option), so if set
-        # via config it will be the same for all relations.
-        return self._all_config_or_relations[0]._namespace
-
-    def _describe_ingresses_action(self, event: Any) -> None:
-        """Handle the 'describe-ingresses' action.
-
-        Args:
-            event: Juju event that fires this handler.
-        """
-        api = self._networking_v1_api()
-        ingresses = api.list_namespaced_ingress(namespace=self._namespace)
-        event.set_results({"ingresses": ingresses})
+        if self.model.relations["nginx-route"]:
+            relation = self.model.relations["nginx-route"][0]
+            if relation.app is None or not relation.data[relation.app]:
+                return None
+            option_essence = IngressOptionEssence.from_nginx_route(
+                self, self.model.relations["nginx-route"][0]
+            )
+        elif self.model.relations["ingress"]:
+            relation = self.model.relations["ingress"][0]
+            if relation.app is None or not relation.data[relation.app]:
+                return None
+            option_essence = IngressOptionEssence.from_ingress(
+                self, self.model.relations["ingress"][0], self._ingress_provider
+            )
```

Comment:
I was wondering if this would be suitable for `state.py` (I've been using relations as state, I might be wrong though!).

Reply:
Since it's just one function for now, I guess we can put it here. If we are expanding this function in the future, we can move this to a different place. Thanks!